internal/terraform/node_resource_abstract_instance.go

```diff
@@ -693,7 +693,7 @@ func (n *NodeAbstractResourceInstance) refresh(ctx EvalContext, deposedKey state
 	}
 
 	// Providers are supposed to return null values for all write-only attributes
-	writeOnlyDiags := ephemeral.ValidateWriteOnlyAttributes(resp.NewState, schema, n.ResolvedProvider, n.Addr)
+	writeOnlyDiags := ephemeral.ValidateWriteOnlyAttributes("Provider produced invalid object", resp.NewState, schema, n.ResolvedProvider, n.Addr)
```

Comment:
Should this message explicitly mention 'refresh'? Here I've just copied the error message used nearby so I think it's appropriate as it is.